app/src/main/java/com/example/adrian/workshop1/Profil.java

```diff
@@ -0,0 +1,149 @@
+                if(response.isSuccessful()) {
+                    updateUI(response.body());
+                } else {
+                    Toast.makeText(Profil.this, "Couldn't fetch profile", Toast.LENGTH_LONG).show();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<ProfileData> call, Throwable t) {
+                t.printStackTrace();
+                Toast.makeText(Profil.this, "Internet problem", Toast.LENGTH_LONG).show();
+            }
+        });
+    }
+
+    private void updateUI(ProfileData data) {
+        Location = (TextView) findViewById(R.id.location);
+        underlined_text(Location, data.getLocation());
+
+        Email = (TextView) findViewById(R.id.email);
+        underlined_text(Email, data.getEmail());
+
+        Name = (TextView) findViewById(R.id.name);
+        if(data.getName() != null)
+            Name.setText(data.getName());
+
+        Employee = (TextView) findViewById(R.id.employee);
+        underlined_text(Employee, data.getCompany());
+
+        Description = (TextView) findViewById(R.id.description);
+        underlined_text(Description, data.getBio());
+
+        Created = (TextView) findViewById(R.id.created);
+        underlined_text(Created, data.getCreatedAt());
+
+        Updated = (TextView) findViewById(R.id.updated);
+        underlined_text(Updated, data.getUpdatedAt());
+
+        PublicRepos = (TextView) findViewById(R.id.public_repo);
+        underlined_text(PublicRepos, data.getPublicRepos().toString());
+
+        PrivateRepos = (TextView) findViewById(R.id.private_repo);
+        underlined_text(PrivateRepos, data.getTotalPrivateRepos().toString());
+    }
+
+    private void underlined_text(TextView v , String string) {
+        if (string != null) {
+            SpannableString content = new SpannableString(string);
+            content.setSpan(new UnderlineSpan(), 0, content.length(), Spanned.SPAN_INCLUSIVE_EXCLUSIVE);
+            v.setText(content);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.action_bar, menu);
+        return true;
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.blog_button:
+                Intent intent = new Intent(Profil.this, RepoActivity.class);
+                startActivity(intent);
+                break;
+        }
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.logout:
+                SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(this);
+                pref.edit().putString("login", null).apply();
+                Intent myIntent = new Intent(Profil.this, PrimulActivity.class);
+                startActivity(myIntent);
+                finish();
+                break;
```

Comment:
Te rog sa faci return true in case-urile pe care le tratezi si return false la sfarsitul functiei onOptionItemSelected. E best practice sa faci asta pentru ca astfel informezi restul aplicatiei ca tu ai facut handling pe actiunile tale, si nu si pe altele.